Quaver.Shared/Screens/Result/UI/ResultHitDifferenceGraph.cs

```diff
@@ -0,0 +1,305 @@
+        /// </summary>
+        /// <param name="screen"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentOutOfRangeException"></exception>
+        private static ScoreProcessor GetScoreProcessor(ResultScreen screen)
+        {
+            // If we already have stats (for example, this is a result screen right after a player finished playing a map), use them.
+            if (screen.ScoreProcessor.Stats != null)
+            {
+                return screen.ScoreProcessor;
+            }
+
+            // Otherwise, get the stats from a replay.
+            Replay replay = null;
+
+            // FIXME: unify this logic with watching a replay from a ResultScreen.
+            try
+            {
+                switch (screen.ResultsType)
+                {
+                    case ResultScreenType.Gameplay:
+                    case ResultScreenType.Replay:
+                        replay = screen.Replay;
+                        break;
+                    case ResultScreenType.Score:
+                        // Don't do anything for online replays since they aren't downloaded yet.
+                        if (!screen.Score.IsOnline)
+                            replay = new Replay($"{ConfigManager.DataDirectory.Value}/r/{screen.Score.Id}.qr");
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+            catch (Exception e)
+            {
+                NotificationManager.Show(NotificationLevel.Error, "Unable to read replay file");
+                Logger.Error(e, LogType.Runtime);
+            }
+
+            // Load a replay if we got one.
+            if (replay != null)
+            {
+                var qua = ResultScreen.Map.LoadQua();
+                MapLoadingScreen.ApplyModsToQua(qua, replay.Mods);
+                var player = new VirtualReplayPlayer(replay, qua);
+                player.PlayAllFrames();
+                return player.ScoreProcessor;
+            }
+
+            // This has no stats, return just as a fallback.
+            return screen.ScoreProcessor;
+        }
+
+        /// <summary>
+        /// Creates the middle line at 0 ms hit difference.
+        /// </summary>
+        /// <param name="middle"></param>
+        private void CreateMiddleLine(int middle)
+        {
+            var _ = new Sprite()
+            {
+                Parent = this,
+                Alpha = 0.7f,
+                Y = middle,
+                Size = new ScalableVector2(Width, 1),
+            };
+        }
+
+        /// <summary>
+        /// Creates the judgement area backgrounds.
+        /// </summary>
+        /// <param name="middle"></param>
+        /// <param name="convertDifferenceToY"></param>
+        private void CreateJudgementAreas(int middle, Func<float, float> convertDifferenceToY)
```

Comment:
Rework this so that the parameters aren't taking in a middle value and a function like this. My previous comments should be clear as to how to go about doing that.